App 2Extended : add HLG support

valnoel opened the pull request:
Depends on https://github.com/Netflix/photon/pull/263

MarcAntoine-Arnaud committed:
declare new colorimetry Color.8DPP - BT2020 HLG

MarcAntoine-Arnaud committed:
if no subdescriptor, get the bitdepth from the componentDepth

MarcAntoine-Arnaud committed:
declare Application TSP 2121

MarcAntoine-Arnaud committed:
fix check ContainerFormat

MarcAntoine-Arnaud committed:
add copyright

valnoel committed:
Add Color.8 support to Application 2 Extended color bit depths map

cconcolato commented on src/main/java/com/netflix/imflibrary/Colorimetry.java:
```diff
@@ -22,6 +22,7 @@
     Color6(ColorPrimaries.P3D65,      TransferCharacteristic.SMPTEST2084,       CodingEquation.None),
     Color7(ColorPrimaries.ITU2020,    TransferCharacteristic.SMPTEST2084,       CodingEquation.ITU2020NCL),
     Color_App5_AP0(ColorPrimaries.ACES,    TransferCharacteristic.Linear,       CodingEquation.None),
+    Color8DPP(ColorPrimaries.ITU2020, TransferCharacteristic.ITU2020HLG,        CodingEquation.ITU2020NCL),
```
Can we use the name as defined in https://github.com/SMPTE/st2067-21-2016-am1/blob/master/35PM-CD-ST-2067-21-2016-AM1.pdf, i.e. simply Color8?

MarcAntoine-Arnaud replied:
I'll change, it's related to firsts DPP documents which references like that. Made sense to update it.

cconcolato commented on src/main/java/com/netflix/imflibrary/st2067_2/ApplicationTsp2121Composition.java:
```diff
@@ -0,0 +1,268 @@
+/*
```
Can you avoid adding TSP 2121 code in the 2067-2 folder? Can you use a separate folder? Also can you clarify why this is not in the other PR https://github.com/Netflix/photon/pull/263 ?

MarcAntoine-Arnaud replied:
Right for a new folder !
This PR is only for the HLG support in App2e, but it requires updates from PR #263 that's why it's a new branch from the TSP 2121, and it includes all commits. If we merge the TSP 2121 first, this PR will contains only the latest commit.

MarcAntoine-Arnaud replied:
And what about the App5 stored also in the App2 folder ?

cconcolato commented on src/main/java/com/netflix/imflibrary/st2067_2/CompositionImageEssenceDescriptorModel.java:
```diff
@@ -384,10 +384,13 @@ public TransferCharacteristic getTransferCharacteristic() {
-                    IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL,
-                    String.format("EssenceDescriptor with ID %s is missing SubDescriptors", imageEssencedescriptorID.toString()));
-
+            Integer componentDepth = getComponentDepth();
```
Section 6.1.2.4.1 of ST-2067-20:2016 indicates: 
"The Top-Level File Package of the Image Track File shall reference a JPEG 2000 Picture Sub Descriptor
SMPTE ST 422 as constrained by Table 13."
So it is an error in App2 and App2e (in the current published editions) to not have a subdescriptor. If this is relaxed, there should be a new namespace, etc.

MarcAntoine-Arnaud replied:
does it means I need to create also a CompositionImageEssenceDescriptorModel for the TSP 2121-1 ?

valnoel committed:
Move the ApplicationTsp2121Composition class to a new tsp_2121 package

valnoel committed:
Rename Color.8 enum value

valnoel committed:
Add specialized CompositionImageEssenceDescriptorModel for TSP 2121

valnoel committed:
Revert CompositionImageEssenceDescriptorModel subdescriptor check on pixel bit depth parsing

valnoel committed:
ApplicationTsp2121Composition handles the new CompositionImageEssenceTsp2121DescriptorModel

And a little refactoring

valnoel committed:
Revert App 2&2E compostion unit tests

valnoel commented:
Hello @cconcolato and @MarcAntoine-Arnaud ,
After your remarks, I've made the changes related to TSP-2121 on the related branch (see #263 ), and rebased and updated this one concerning the HLG support in Application 2E.
Thanks for your review.
